src/context-selection/use-context-selection/use-is-valid-selection.js

```diff
@@ -1,12 +1,31 @@
-import { useAttributeParams } from '../attribute-option-combo-selector-bar-item/index.js'
+import { useMetadata, selectors } from '../../metadata/index.js'
 import { useContextSelection } from '../use-context-selection/index.js'
 
 export function useIsValidSelection() {
-    const [selection] = useContextSelection()
-    const { validSelection } = useAttributeParams()
+    const [{ dataSetId, orgUnitId, periodId, attributeOptionComboSelection }] =
+        useContextSelection()
+    const { data } = useMetadata()
 
-    if (selection.orgUnitId && selection.periodId && validSelection) {
+    let validAttributeComboSelection = false
+    if (data && dataSetId) {
+        const dataSet = selectors.getDataSetById(data, dataSetId)
+        const catComboId = dataSet.categoryCombo.id
+        const categoryCombo = selectors.getCategoryComboById(data, catComboId)
+
+        const selectedOptions = Object.values(attributeOptionComboSelection)
+
+        // if default catCombo, no selection is needed
+        if (categoryCombo.isDefault) {
+            validAttributeComboSelection = true
+        } else {
+            validAttributeComboSelection =
+                categoryCombo.categories.length === selectedOptions.length
+        }
+    }
+
+    if (orgUnitId && periodId && validAttributeComboSelection) {
         return true
     }
+
     return false
```

Comment:
I don't think we need to check whether `data` exists or not, the `<LoadApp />` component takes care of that, right?

```suggestion
    if (!dataSetId || !orgUnitId || !periodId) {
        return false
    }
    
    const dataSet = selectors.getDataSetById(data, dataSetId)
    const catComboId = dataSet.categoryCombo.id
    const categoryCombo = selectors.getCategoryComboById(data, catComboId)
    
    if (categoryCombo.isDefault) {
        return true
    }

    const selectedOptions = Object.values(attributeOptionComboSelection)
    return categoryCombo.categories.length === selectedOptions.length
```